### DISCUSSION TITLE
How to get YJIT stats in web server processes?

### DISCUSSION THREAD

**Discussion by PikachuEXE** [puma/puma#3321] [Q&A] [answered]
Upgraded to Ruby 3.3, trying to determine what memory limit to set (via `--yjit-exec-mem-size`, see https://github.com/ruby/ruby/blob/master/doc/yjit/yjit.md)

Though in the same doc it's suggested to use `RubyVM::YJIT.runtime_stats[:ratio_in_yjit]` to get `the ratio of YJIT-executed instructions in %`. But how to run that on a running server process (like puma)?

Also asked in ruby subreddit
https://old.reddit.com/r/ruby/comments/19ewrpb/how_to_get_yjit_stats_in_web_server_processes/

**Comment by dentarg** [comment, marked as answer]
Maybe the out of band hook can be used? https://github.com/puma/puma/blob/f6ab4e9b5fd3809acaab9407a79da4add7e77e31/lib/puma/dsl.rb#L789

**Reply by PikachuEXE** [reply]
Thanks
Due to a bug in ruby 3.3.0 I can't probably test it yet
[Ruby 3.3.0 ignores --yjit-stats when RubyVM::YJIT.enable (no argument) is used, which is an unintended behavior.](https://bugs.ruby-lang.org/issues/20153)

**Reply by PikachuEXE** [reply]
It seems the bug can be workaround via 
- `RUBYOPT="--yjit-disable --yjit-exec-mem-size=128"` (no `--yjit-stats=quiet`)
-  `RubyVM::YJIT.enable(stats: :quiet)`
